src/components/taskDetails/TaskDates.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import Details from './Details';
+import styles from './task-details.module.scss';
+import { TASK_EXTENSION_REQUEST_URL } from '@/constants/url';
+import convertTimeStamp from '@/helperFunctions/convertTimeStamp';
+
+interface TaskDatesProps {
+    isEditing: boolean;
+    isUserAuthorized: boolean;
```

Comment:
are these all required property?

Reply:
- `isEditing`: Controls the rendering of the input field for editing the `endsOn` date.
- `isUserAuthorized`: Ensures that only authorized users can edit the `endsOn` date.
- `startedOn`: Displays the start date of the task, which is crucial information for the user.
- `endsOn`: Displays the end date of the task, which is also essential for the task timeline.
- `newEndOnDate`: Holds the current value of the edited endsOn date, allowing for controlled input.
- `setNewEndOnDate`: A callback function needed to update the `newEndOnDate` state when the user modifies the date input.
- `handleBlurOfEndsOn`: Ensures that the new end date is saved when the user leaves the date input field.
- `isExtensionRequestPending`: Determines whether a task's extension request is pending, which influences the URL rendered for the `Ends On` date.
- `taskId`: Provides the unique identifier for the task, necessary for constructing the URL for extension requests.